src/bvals/comms/bnd_info.cpp

```diff
@@ -206,6 +208,22 @@ SpatiallyMaskedIndexer6D CalcIndices(const NeighborBlock &nb, MeshBlock *pmb,
     }
   }
 
+  // Transform to logical coordinates of neighbor block if this
+  // is a receiving block
+  if (ir_type == IndexRangeType::BoundaryExteriorRecv) {
+    s = lcoord_trans.Transform(s);
+    e = lcoord_trans.Transform(e);
+    // Transformation can flip the order of the upper and
+    // lower index, so make sure they are increasing
+    for (int dir = 0; dir < 3; ++dir) {
+      if (s[dir] > e[dir]) {
+        int temp = s[dir];
+        s[dir] = e[dir];
+        e[dir] = temp;
+      }
+    }
+  }
```

Comment:
I think the index calculation has almost no impact on performance, so I doubt this will change that. It is certainly possible that calling the transformation in the set boundaries kernel will have some performance impact. If this is the case, we can always have two sets of inner loops w/ and w/o the transformation and switch between them based on if the transformation is trivial or not.

Reply:
:+1: I think the two loop solution is reasonable. And I don't think we need to do anything now. I want us to have an eye on this though to avoid a mystery performance regression.